src/main/java/org/jenkinsci/plugins/kubernetes/auth/KubernetesAuthCertificate.java

```diff
@@ -22,8 +26,26 @@ public KubernetesAuthCertificate(String certificate, String key) {
     }
 
     @Override
-    public void decorate(AuthInfoBuilder builder) {
-        builder.withClientCertificateData(certificate).withClientKeyData(key);
+    public String buildKubeConfig(String serverUrl, String caCertificate) throws JsonProcessingException {
```

Comment:
Create an abstract class and put all the common things in there to avoid duplicating code.